lib/bot.js

```diff
@@ -558,6 +558,46 @@ module.exports = (function() {
 		},
 
 
+		addFlowTopic(title, subject, content, callback) {
+			const self = this;
+
+			if (this.dryRun) {
+				callback(new Error('In dry-run mode'));
+				return;
+			}
+
+			// @see http://www.mediawiki.org/wiki/API:Flow
+			this.getToken(title, 'flow', function(err, token) {
+				if (err) {
+					callback(err);
+					return;
+				}
+
+				self.log(`Adding a topic to page '${title}' with subject '${subject}'...`);
+
+				const params = {
+					action: 'flow',
+					submodule: 'new-topic',
+					page: title,
+					nttopic: subject,
+					ntcontent: content,
+					ntformat: 'wikitext',
+					bot: '',
+					token: token
+				};
+
+				self.api.call(params, function(err, data) {
+					if (!err && data['new-topic'] && data['new-topic'].status && data['new-topic'].status === "ok") {
+						self.log("Workflow '%s' created on '%s'", data['new-topic'].workflow, title);
+						callback(null, data);
+					}
+					else {
+						callback(new Error(`Edit failed: ${err}`));
```

Comment:
I'd simply pass `err` (without creating a new exception object) to the callback to be consistent with the rest of the code.